staging/src/k8s.io/apimachinery/pkg/util/diff/diff.go

```diff
@@ -17,122 +20,53 @@ limitations under the License.
 package diff
 
 import (
-	"bytes"
+	"encoding/json"
 	"fmt"
 	"reflect"
-	"strings"
-	"text/tabwriter"
 
-	"github.com/google/go-cmp/cmp" //nolint:depguard
-	"k8s.io/apimachinery/pkg/util/dump"
+	"github.com/pmezard/go-difflib/difflib"
 )
 
-func legacyDiff(a, b interface{}) string {
-	return cmp.Diff(a, b)
-}
-
-// StringDiff diffs a and b and returns a human readable diff.
-// DEPRECATED: use github.com/google/go-cmp/cmp.Diff
-func StringDiff(a, b string) string {
-	return legacyDiff(a, b)
-}
+// Diff returns a string representation of the difference between two objects.
+// When built with the nogocmp tag, it uses difflib to generate a unified diff
+// of the JSON representation of the objects.
```

Comment:
does this doc need updating since the tag flipped polarity and now defaults to this unless opting back into gocmp via usegocmp?

Reply:
whoops! yes